### PR TITLE
PI-2472 - Fix to Resolve flaky OASys End2End Tests and failures in the pipeline & increate slow time for Delius case data test

### PR DESCRIPTION
**Pull request by smalepati99**
This pull request focuses on enhancing the stability and reliability of the OASYS Tests. The changes aim to address issues causing test failures and flakiness, thereby improving the overall performance and consistency of the pipeline.

1. Refactored Section Clicks by modifying functions for clicking on sections within the OASYS application to ensure that interactions are more reliable and less prone to failure.
2. Updated clickSection2to13 Function by improving the logic to handle the needs parameter more effectively, ensuring that the flow is consistent with the specified needs ('Yes' or 'No').
3. Enhanced Optional Parameters by adjusting function signatures to handle optional parameters more robustly, especially in scenarios where certain parameters might be undefined. This includes updating how needs is managed in various functions.
4. Increased Timeout and Error Handling by updating timeout settings and error handling mechanisms to better manage scenarios where elements may not be immediately available, reducing the likelihood of test failures due to timing issues.
5. Code Cleanup by removing redundant or obsolete code, ensuring that the test suite is cleaner and easier to maintain.
6. Consistency Improvements by standardising the approach to handling various test scenarios, leading to more predictable and stable test outcomes.

**These changes are expected to:**

- Reduce Test Failures by improving interaction methods and handling of optional parameters, the likelihood of test failures due to intermittent issues will be minimized.
- Eliminate Flakiness because of the adjustments to timeouts and error handling will contribute to a more stable and consistent test execution, reducing flaky test behavior.
- Improve Maintenance because of the cleaner and more consistent code will make it easier for developers to maintain and extend the test suite in the future.

### CODE REVIEW

File: steps/oasys/layer3-assessment/create-assessment.ts [hunk 1]
```diff
@@ -307,9 +306,27 @@ export const clickEducationTrainingEmpl = async (page: Page) => {
...
+        async () => {
+            if (!(await linkLocator.isVisible())) {
+                await sectionLocator.click()
+            }
+        },
+        async () => {
+            await expect(linkLocator).toBeVisible()
+        },
+        { timeout: 60_000, intervals: [500, 1000, 5000] }
+    )
```
Comment (marcus-bcl) [inline steps/oasys/layer3-assessment/create-assessment.ts]: Do you need the `doUntil` here?  Would this have the same result:

```suggestion
    if (!(await linkLocator.isVisible())) {
        await sectionLocator.click()
    }
    await expect(linkLocator).toBeVisible()
```

File: steps/oasys/layer3-assessment/create-layer3-assessment/add-layer3-needs.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { completeRelationshipsSection } from '../relationships-section'
 import { signAndlock } from '../sign-and-lock'
 
 export const addLayer3AssessmentNeeds = async (page: Page, role?: string) => {
-    await page.locator('a', { hasText: 'Section 2 to 13' }).click()
+    // await page.locator('a', { hasText: 'Section 2 to 13' }).click()
```
Comment (marcus-bcl) [inline steps/oasys/layer3-assessment/create-layer3-assessment/add-layer3-needs.ts]: Is this line still needed?

File: steps/oasys/layer3-assessment/create-assessment.ts (hunk 1 above)
Comment (smalepati99) [reply steps/oasys/layer3-assessment/create-assessment.ts]: I have changed now.

File: steps/oasys/layer3-assessment/create-layer3-assessment/add-layer3-needs.ts (hunk 2 above)
Comment (smalepati99) [reply steps/oasys/layer3-assessment/create-layer3-assessment/add-layer3-needs.ts]: Removed now

**Review by pmcphee77 — APPROVED** [review]

**Review by pmcphee77 — APPROVED** [review]

**Pull request merged**